vm/src/types/slot.rs

```diff
@@ -41,7 +41,7 @@ pub struct PyTypeSlots {
     // Methods to implement standard operations
 
     // Method suites for standard classes
-    pub as_number: AtomicCell<Option<PointerSlot<PyNumberMethods>>>,
+    pub as_number: Option<&'static PyNumberMethods>,
```

Comment:
Because every PyPayload has `PyNumberMethods` - or even NOT_IMPLEMENTED table, this field can be non-optional. 
```suggestion
    pub as_number: &'static PyNumberMethods,
```